src/core/render/index.js

```diff
@@ -48,22 +46,35 @@ function renderMain(html) {
   }
 
   this._renderTo('.markdown-section', html);
+
   // Render sidebar with the TOC
   !this.config.loadSidebar && this._renderSidebar();
 
-  // Execute script
-  if (
-    this.config.executeScript !== false &&
-    typeof window.Vue !== 'undefined' &&
-    !executeScript()
-  ) {
-    setTimeout(_ => {
-      const vueVM = window.__EXECUTE_RESULT__;
-      vueVM && vueVM.$destroy && vueVM.$destroy();
-      window.__EXECUTE_RESULT__ = new window.Vue().$mount('#main');
-    }, 0);
-  } else {
-    this.config.executeScript && executeScript();
+  // Execute markdown <script>
+  if (this.config.executeScript || 'Vue' in window) {
+    executeScript();
+  }
+
+  // Handle Vue content not handled by markdown <script>
+  if ('Vue' in window) {
```

Comment:
This entire `if ('Vue' in window)` block exists to maintain the current behavior of handling basic Vue rendering without requiring a markdown `<script>` tag. My preference would be to remove this block in v5 and just require users to include `<script>` file in pages that contain Vue content instead of docsify assuming every page has Vue content.

Reply:
We need to double check this and add unit tests for the specific behavior we want.

The behavior here was bad before, it was executing script tags twice due to the two calls to `executeScript`. 

I was planning to to circle back to this particular piece of code based on changes in https://github.com/docsifyjs/docsify/pull/1091 (please see the last item in the TODO list there).

Reply:
@trusktr 

> The behavior here was bad before [...]

Not sure which behavior you're referring to, but I agree there are multiple aspects of the current behavior that should be revisited (see below). That said, my goal with this PR is to provide a fix (actually, an enhancement since this dynamic Vue content has never worked as far as I can tell) without a breaking change.

>  [...] it was executing script tags twice due to the two calls to executeScript

I'm not seeing this behavior in the latest version (4.11.4), verified (manually) using the following demo: https://codesandbox.io/s/docsify-790-9o1w8?file=/index.html

- When Vue.js is **not** detected, markdown `<script>`...
  - is not executed when `executeScript` is `false`
  - is not executed when `executeScript` is falsy or unspecified
  - is executed when `executeScript` is `true`
  - is executed when `executeScript` is truthy
- When Vue.js **is** detected, markdown `<script>`...
  - is not executed when `executeScript` is `false`
  - ⚠️ is executed when `executeScript` is falsy or unspecified
  - is executed when `executeScript` is `true`
  - is executed when `executeScript` is truthy

Markdown `<script>` tags are not being executed twice in any of the above scenarios. 

The ⚠️ icon indicates a what I believe to be a problematic behavior, however it is one that should remain in order to prevent a breaking change. This behavior is as described docsify's [`executeScript` documentation](https://docsify.js.org/#/configuration?id=executescript) documentation (although the phrasing is a little confusing):

  > ## executeScript
  > type: `Boolean`
  > Execute the script on the page. Only parse the first script tag(demo). If Vue is present, it is turned on by default.

  Note that I have made an update to this PR to match the above behavior (specifically, to allow `executeScript:false` to prevent markdown `<script>` from executing even if Vue is detected). Proof that testing is needed, so you'll get no arguments from me. :)

> We need to double check this and add unit tests for the specific behavior we want.

Yes we should add tests, but the behavior we want is the same behavior above _and_ the ability to render dynamic Vue content. Just clarifying that "the specific behavior we want" we want now and down the road may be different.

> I was planning to to circle back to this particular piece of code based on changes in #1091 (please see the last item in the TODO list there).

One less thing to do!